scripts/environment/gcloud_env.py

```diff
@@ -0,0 +1,219 @@
+# Copyright 2016 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Set up the gcloud environment and create a new project with App Engine.
+#
+# This has been tested with python 2.7.
+
+import re
+import time
+
+from distutils.spawn import find_executable
+from subprocess import PIPE, Popen, call
+
+class GcloudEnvironment(object):
+    """
+    Encapsulate the gcloud setup in the GcloudEnvironment class
+    """
+
+    PROJECT_ID_REGEX = re.compile('^[a-z][a-z0-9-]{6,30}$')
+
+    def __init__(self):
+        self.config_name = None
+        self.auth_account = None
+        self.project_id = None
+        self.region = None
+
+    def __repr__(self):
+        return ('GcloudEnvironment:\n'
+                '  Account: {}\n'
+                '  ProjectId: {}\n'
+                .format(self.auth_account,
+                        self.project_id))
+
+    def ensure_installed(self):
+        """
+        Check whether gcloud tool is installed.
+        """
+        gcloud_cmd = find_executable('gcloud')
+        if gcloud_cmd:
+            print 'Found gcloud tool!'
+        else:
+            raise EnvironmentError(
+                'Could not find gcloud. '
+                'Have you installed the Google Cloud SDK?\n'
+                'You can get it here: https://cloud.google.com/sdk/')
+
+    def auth_login(self):
+        """
+        User needs to authenticate with Google Cloud Platform account
+        before doing anything else.
+        """
+        return_val = call(['gcloud', 'auth', 'login', '--force'])
+        p = Popen(['gcloud', 'auth', 'list',
+                   '--filter=status:ACTIVE', '--format=value(account)'],
+                  stdout=PIPE, stderr=PIPE)
+        stdout, stderr = p.communicate()
+        try:
+            self.account = stdout.strip()
+        except:
```

Comment:
ditto, usually prefer to catch targetted exceptions, rather than a generic catch-all